File: jvm-langs/app/src/main/java/org/javacode/Exercism.java
package org.javacode;

public class Exercism {

    public static String reverseString(String inputStr) {
        char[] charArray = inputStr.toCharArray();
        int left = 0, right = charArray.length - 1;

        while(left < right) {
            char temp = charArray[left];
            charArray[left] = charArray[right];
            charArray[right] = temp;

            left++;
            right--;
        }

        return new String(charArray);
    }
}

File: app/bin/main/org/kotlincode/Exercism.kt
package org.kotlincode

object Exercism {

    fun reverseString(inputStr: String): String {
        val charArray = inputStr.toCharArray()
        var leftPointer = 0
        var rightPointer = charArray.lastIndex

        while (leftPointer < rightPointer) {
            val tmp = inputStr[leftPointer]
            charArray[leftPointer] = charArray[rightPointer]
            charArray[rightPointer] = tmp

            leftPointer++
            rightPointer++
        }

        return String(charArray)
    }
}